tests/beta/tapes/test_qnode.py

```diff
@@ -0,0 +1,429 @@
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit tests for the QNode"""
+import pytest
+import numpy as np
+
+import pennylane as qml
+from pennylane.beta.tapes import (
+    QuantumTape,
+    QNode,
+    qnode,
+    QuantumFunctionError
+)
+from pennylane.beta.queuing import expval, var, sample, probs, MeasurementProcess
+
+
+class TestValidation:
+    """Tests for QNode creation and validation"""
+
+    def test_invalid_interface(self):
+        """Test that an exception is raised for an invalid interface"""
+        dev = qml.device("default.qubit", wires=1)
+
+        with pytest.raises(QuantumFunctionError, match="Unknown interface"):
+            QNode(None, dev, interface="something")
+
+    def test_invalid_device(self):
+        """Test that an exception is raised for an invalid device"""
+        with pytest.raises(QuantumFunctionError, match="Invalid device"):
+            QNode(None, None)
+
+    def test_validate_device_method(self, monkeypatch):
+        """Test that the method for validating the device diff method
+        tape works as expected"""
+        dev = qml.device("default.qubit", wires=1)
+
+        with pytest.raises(
+            QuantumFunctionError,
+            match="does not provide a native method for computing the jacobian",
+        ):
+            QNode._validate_device_method(dev, None)
+
+        monkeypatch.setitem(dev._capabilities, "provides_jacobian", True)
+        tape_class, interface, method = QNode._validate_device_method(dev, "interface")
+
+        assert tape_class is QuantumTape
+        assert method == "device"
+        assert interface == "interface"
+
+    def test_validate_backprop_method(self, monkeypatch):
+        """Test that the method for validating the backprop diff method
+        tape works as expected"""
+        dev = qml.device("default.qubit", wires=1)
+
+        with pytest.raises(QuantumFunctionError, match="does not support native computations"):
+            QNode._validate_backprop_method(dev, None)
+
+        test_interface = "something"
+        monkeypatch.setitem(dev._capabilities, "passthru_interface", test_interface)
+
+        with pytest.raises(QuantumFunctionError, match=f"when using the {test_interface}"):
+            QNode._validate_backprop_method(dev, None)
+
+        tape_class, interface, method = QNode._validate_backprop_method(dev, test_interface)
+
+        assert tape_class is QuantumTape
+        assert method == "backprop"
+        assert interface == None
+
+    def test_get_parameter_shift_method(self, monkeypatch):

```

Comment:
Similar thoughts for splitting up (this one seems a better case for that).

Reply:
done!